src/utils/utils.ts

```diff
@@ -1117,3 +1118,51 @@ export const validConcentrationMidPriceTick = (
 
   return midPriceTick
 }
+
+type LiquidityTickWithoutIndex = Omit<LiquidityTick, 'index'>
+
+export const getLiquidityTicksByPositionsList = (
+  poolKey: PoolKey,
+  positions: Position[]
+): LiquidityTick[] => {
+  const ticks: Record<number, LiquidityTickWithoutIndex> = {}
+
+  positions.forEach(position => {
+    if (poolKeyToString(position.poolKey) === poolKeyToString(poolKey)) {
+      updateTick(ticks, Number(position.lowerTickIndex), position.liquidity, true)
+      updateTick(ticks, Number(position.upperTickIndex), position.liquidity, false)
+    }
+  })
+
+  return Object.entries(ticks).map(([index, { liquidityChange, sign }]) => ({
+    index: BigInt(index),
+    liquidityChange,
+    sign
+  }))
+}
+
+export const updateTick = (
+  ticks: Record<number, LiquidityTickWithoutIndex>,
+  tickIndex: number,
+  positionLiquidity: bigint,
+  isLowerTick: boolean
+) => {
+  if (!ticks[tickIndex]) {
+    ticks[tickIndex] = {
```

Comment:
shouldn't you stop at this action if the tick wasn't found? Isn't it the case that you would then add or subtract liquidity twice?

Reply:
If this should be initialized as empty tick, the fields will look like this:
```Typescript
      liquidityChange: 0n
      sign: `true` // regardless of whether it's -0 or +0
```